client/lib/pages/add-plant.dart

```diff
@@ -13,14 +15,106 @@ class AddPlant extends StatefulWidget {
 class _AddPlant extends State<AddPlant> {
   final _formKey = GlobalKey<FormState>();
   final UserPlant newPlant = new UserPlant();
+  // TODO: add text
+  final _distanceToWindowText = <String>['Verweg', '2', '3', '4', 'Dichtbij'];
+
+  String selectedImagePath;
+  DateTime selectedDate;
+  TimeOfDay selectedTime;
   int minTemp;
   int maxTemp;
 
+  // make date + time picker optional
+  bool hideDatePicker = false;
+  bool showFutureTimeWarning = false;
+  bool allowedToSubmit = false;
+
+  // dd-MM-yyyy HH:mm
+  String formatDate(DateTime date) {
+    if (date == null) {
+      throw new ArgumentError('date is null');
+    }
+
+    var year = newPlant.lastWaterDate.year;
+    var month = newPlant.lastWaterDate.month.toString().padLeft(2, '0');
+    var day = newPlant.lastWaterDate.day.toString().padLeft(2, '0');
+    var hour = newPlant.lastWaterDate.hour.toString().padLeft(2, '0');
+    var minute = newPlant.lastWaterDate.minute.toString().padLeft(2, '0');
+
+    return '$day-$month-$year $hour:$minute';
+  }
+
+  void selectImageFromSource(BuildContext context, ImageSource source) async {
+    var pickedImage = await ImagePicker.pickImage(source: source);
+    Navigator.of(context).pop();
+
+    setState(() {
+      // keep the previous image if no image is selected
+      if (pickedImage == null && selectedImagePath != null) {
+        return;
+      }
+
+      selectedImagePath = pickedImage != null ? pickedImage.path : null;
+    });
+  }
+
   void submit() {
-    if (this._formKey.currentState.validate()) {
+    // TODO: invalidate if given lastWaterDate is null and showFutureTimeWarning is true
+    if (this._formKey.currentState.validate() && allowedToSubmit) {
       _formKey.currentState.save();
+      newPlant.imageName = selectedImagePath;
       User.plants.add(newPlant);
+      Navigator.pushNamed(context, '/my-plants');
+    }
+  }
+
+  void pickDate(BuildContext context) async {
+    DateTime picked = await showDatePicker(
+        context: context,
+        initialDate: selectedDate ?? DateTime.now(),
+        firstDate: DateTime(DateTime.now().year - 1, 1, 1),
+        lastDate: DateTime.now());
+
+    if (picked != null) {
+      setState(() => {selectedDate = picked});
+      print('picked date: $selectedDate');
+      _combinePickedDateAndTime();
+    }
+  }
+
+  void pickTime(BuildContext context) async {
+    TimeOfDay picked = await showTimePicker(
+      context: context,
+      initialTime: selectedTime ?? TimeOfDay.now(),
+    );
+
+    if (picked != null) {
+      setState(() => {selectedTime = picked});
+      print('picked time: $selectedTime');
+      _combinePickedDateAndTime();
+    }
+  }
```

Comment:
Can't this be in one button. You press the button it will then show the date first and the time right after. Also you can still add the plant even if the time is in the future. You are checking on this but not good enough.